### PR TITLE
Refactor dialer package

### PR DESCRIPTION
**Pull request by jxx-gg**
It consolidates with the dialer package in https://github.com/openconfig/gnmi/tree/master/tunnel/dialer.

### CODE REVIEW

File: dialer/dialer.go [hunk 1]
```diff
@@ -55,15 +74,20 @@ func FromClient(tc *tunnel.Client) (*ClientDialer, error) {
 //
 // conn, err := d.DialContext(ctx, "target1", "target-type1", opts1)
 // conn, err := d.DialContext(ctx, "target2", "target-type2", opts2)
-func (d *ClientDialer) DialContext(ctx context.Context, target, targetType string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+func (cd *ClientDialer) DialContext(ctx context.Context, target, targetType string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
```
Comment (gcsl) [inline dialer/dialer.go:77]: Receiver can be just d.

File: dialer/dialer.go [hunk 2]
```diff
@@ -55,15 +74,20 @@ func FromClient(tc *tunnel.Client) (*ClientDialer, error) {
...
-		return &tunnel.Conn{session}, nil
+		return tunnel.ClientConn(ctx, cd.tc, &tunnel.Target{ID: target, Type: targetType})
 	})
 	opts = append(opts, withContextDialer)
 	return grpc.DialContext(ctx, target, opts...)
 }
 
+// DialContext establishes a grpc.Conn to the attached tunnel server and
+// returns an error if the connection is not established.
+func (sd *ServerDialer) DialContext(ctx context.Context, target string, targetType string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
```
Comment (gcsl) [inline dialer/dialer.go:87]: Receiver can be just d.

**Review by gcsl — CHANGES_REQUESTED** [review]

File: dialer/dialer.go [hunk 3]
```diff
@@ -41,11 +46,25 @@ type ClientDialer struct {
...
+		return nil, fmt.Errorf("tunnel client connection is nil")
 	}
 	return &ClientDialer{tc: tc}, nil
 }
 
+// FromServer creates a new collector dialer with an existing tunnel server
+// connection.
+// Sample code with error handling elided:
+//
+// conn, err := grpc.DialContext(ctx, tunnelAddress)
```
[the remark this answers is not in the document]
Comment (jxx-gg) [reply dialer/dialer.go]: updated

File: dialer/dialer.go (hunk 1 above)
Comment (jxx-gg) [reply dialer/dialer.go]: done

File: dialer/dialer.go (hunk 2 above)
Comment (jxx-gg) [reply dialer/dialer.go]: done

**Review by gcsl — APPROVED** [review]

**Pull request merged**